cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/elements/AbstractTransformerConversion.java

```diff
@@ -0,0 +1,818 @@
+    }
+
+    private TapChangerConversion moveTapChanger(TapChangerConversion tc) {
+        TapChangerConversion tapChanger = baseCloneTapChanger(tc);
+        moveTapChangerSteps(tapChanger, tc);
+        return tapChanger;
+    }
+
+    private void moveTapChangerSteps(TapChangerConversion tapChanger, TapChangerConversion tc) {
+        tc.getSteps().forEach(step -> {
+            double ratio = step.getRatio();
+            double angle = step.getAngle();
+            double r = step.getR();
+            double x = step.getX();
+            double g1 = step.getG1();
+            double b1 = step.getB1();
+            double g2 = step.getG2();
+            double b2 = step.getB2();
+            TapChangerStepConversion convertedStep = calculateConversionStep(ratio, angle, r, x, g1, b1, g2, b2);
+            tapChanger.beginStep()
+                .setRatio(convertedStep.ratio)
+                .setAngle(convertedStep.angle)
+                .setR(convertedStep.r)
+                .setX(convertedStep.x)
+                .setG1(convertedStep.g1)
+                .setB1(convertedStep.b1)
+                .setG2(convertedStep.g2)
+                .setB2(convertedStep.b2)
+                .endStep();
+        });
+    }
+
+    private TapChangerStepConversion calculateConversionStep(double ratio, double angle, double r,
+        double x, double g1, double b1, double g2, double b2) {
+        TapChangerStepConversion step = new TapChangerStepConversion();
+        Complex a = new Complex(ratio * Math.cos(Math.toRadians(angle)),
+            ratio * Math.sin(Math.toRadians(angle)));
+        Complex na = a.reciprocal();
+        step.ratio = na.abs();
+        step.angle = Math.toDegrees(na.getArgument());
+        step.r = 100 * (impedanceConversion(1 + r / 100, a) - 1);
+        step.x = 100 * (impedanceConversion(1 + x / 100, a) - 1);
+        step.g1 = 100 * (admittanceConversion(1 + g1 / 100, a) - 1);
+        step.b1 = 100 * (admittanceConversion(1 + b1 / 100, a) - 1);
+        step.g2 = 100 * (admittanceConversion(1 + g2 / 100, a) - 1);
+        step.b2 = 100 * (admittanceConversion(1 + b2 / 100, a) - 1);
+        return step;
+    }
+
+    protected RatioConversion identityRatioConversion(double r, double x, double g1,
+        double b1, double g2, double b2) {
+        RatioConversion ratio = new RatioConversion();
+        ratio.r = r;
+        ratio.x = x;
+        ratio.g1 = g1;
+        ratio.b1 = b1;
+        ratio.g2 = g2;
+        ratio.b2 = b2;
+        return ratio;
+    }
+
+    protected RatioConversion moveRatioFrom2To1(double a0, double angle, double r, double x, double g1,
+        double b1, double g2, double b2) {
+        return moveRatio(a0, angle, r, x, g1, b1, g2, b2);
+    }
+
+    protected RatioConversion moveRatioFrom1To2(double a0, double angle, double r, double x, double g1,
+        double b1, double g2, double b2) {
+        return moveRatio(a0, angle, r, x, g1, b1, g2, b2);
+    }
```

Comment:
To convert from the interpreted model to the IIDM model is never used as tapChangers are always moved  from end2 to end1. I included to have both possibilities, maybe it will be necessary later. Should i delete it?